pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+                           'valid CIMQualifierDeclaration: %s' % qual)
+
+        if qual.name in self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_ALREADY_EXISTS, '%s' % qual.name)
+        try:
+            self.qualifiers[namespace][qual.name] = qual
+        except KeyError:
+            self.qualifiers[namespace] = NocaseDict({qual.name: qual})
+
+    def _fake_deletequalifier(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.DeleteQualifier`
+
+        Deletes a single qualifier if it is in the
+        repository for this class and namespace
+
+        Exceptions;
+            CIMError CIM_ERR_INVALID_NAMESPACE, CIMError(CIM_ERR_NOT_FOUND
+        """
+        if not self.qualifiers[namespace]:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
+
+        qname = params['QualifierName']
+
+        if qname in self.qualifiers[namespace]:
+            del self.qualifiers[namespace][qname]
+        else:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Qualifier decl %s not found.'
+                           % qname)
+
+    #####################################################################
+    #
+    #  Faked WBEMConnection Instance methods
+    #
+    #####################################################################
+
+    def _fake_createinstance(self, namespace, **params):
+        """
+        Implements a server reactor for
+        :meth:`~pywbem.WBEMConnection.CreateInstance`
+
+        Create a CIM instance in the local repository of this class.
+
+        Exceptions:
+            CIMError CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_CLASS
+        """
+
+        new_inst = params['NewInstance']
+
+        # Requires corresponding class to build path to be returned
+        if not self._class_exists(namespace, new_inst.classname):
+            raise CIMError(CIM_ERR_INVALID_CLASS, 'Cannot create instance '
+                                                  'without corresponding class '
+                                                  '%s.' % new_inst.classname)
+        # TODO change this to use std getclass
+        class_for_inst = self.classes[namespace][new_inst.classname]
+
+        # If property not in instance, add it and use default value from
+        # class
+        for cprop_name in class_for_inst.properties:
+            if cprop_name not in new_inst:
+                default_value = class_for_inst.properties[cprop_name]
+                new_inst[cprop_name] = default_value
+
+        # Exception if property in instance but not class
+        for iprop in new_inst:
+            if iprop not in class_for_inst.properties:
+                raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                               'Property %s in instance but not in class %s'
+                               % (iprop, class_for_inst))
```

Comment:
1. I would prefer an exception message like "Property %s specified in NewInstance is not exposed by class %s in namespace %s."
2. Mention the namespace name in the exception message.

Reply:
Done